app/models/user.rb

```diff
@@ -0,0 +1,67 @@
+class User < ApplicationRecord
+  SPECIAL = "?<>',?[]}{=-)(*&^%$#`~{}!"
+  REGEX = /[#{SPECIAL.gsub(/./){|char| "\\#{char}"}}]/
+  ROLLER = { user: 0, admin: 1 }
+
+  scope :staff, -> {where(role: User.roles[:user])} 
+
+  validates :first_name, :last_name, presence: true 
+  validates :password, format: { with: /(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[^\w\s]).{8,}/ }
+  validate :password_lower_case, :password_uppercase, :password_special_char, :password_contains_number
+
+  before_create :set_password, :block_from_invitation?
+
+  attr_writer :login
+
+  def login
+    @login || self.username || self.email
```

Comment:
What is the purpose of `@login`?

Reply:
I can use login in rails console to write login by either username or email